### ISSUE TITLE
Console crashes when using column comparison operators in relationship in exists clause of permission check

### ISSUE THREAD

**Issue by BenoitRanque** [hasura/graphql-engine#9456]
### Version Information

Server Version: 2.17.0

### Environment

OSS

### What is the current behaviour?

When attempting to create a permission with a column comparison operator in an array relationship in an exists clause, the console will crash.

### What is the expected behaviour?

Console should not crash

### How to reproduce the issue?

1. Using the chinook schema, attempt to create permissions on the Track table with the following JSON:
```json
{"_exists":{"_table":{"schema":"public","name":"Artist"},"_where":{"Albums":{"Title":{}}}}}
```
2. Attempt to add a column comparison operator
4. Console crashes. Interestingly this does not seem to happen for object relationships.

![image](https://user-images.githubusercontent.com/25712958/220552259-c47de16d-0f80-479f-a4da-6f31179b1539.png)

**Comment by BenoitRanque** [comment]
This makes it harder to work around #9446

**Comment by RodriguezLucha** [comment]
Just ran into this as well! Was upgrading from 2.12 -> 2.17.1 and every time a change was made to the metadata the server would crash. We had row level filtering using exists + where on 3 of our tables and after removing that the crashes stopped

**Comment by ajoyoommen** [comment]
Facing the same issue here

**Comment by vijayprasanna13** [comment]
@ejkkan can we confirm if this issue has been fixed in the latest community version?

**Comment by ejkkan** [comment]
Hey!
I'm closing this as it's fixed in v2.21.x, let us know if you still face this issue
